ocaml/xapi/helpers.ml

```diff
@@ -43,6 +43,27 @@ let log_exn_continue msg f x =
 
 type log_output = Always | Never | On_failure
 
+let filter_patterns =
+  [
+    ( Re.Str.regexp "^\\(.*proxy_\\(username\\|password\\)=\\)\\(.*\\)$"
+    , "\\1(filtered)"
+    )
+  ]
+
+let filter_args args =
+  List.map
+    (fun arg ->
+      List.fold_left
+        (fun acc (r, t) ->
+          if Re.Str.string_match r acc 0 then
```

Comment:
Re.Str is not thread-safe, I thought we fixed that, but perhaps not. We should use Re.Posix or Re.Pcre